mirgecom/diffusion.py

```diff
@@ -175,44 +214,67 @@ def __init__(self, value):
         """
         self.value = value
 
-    def get_grad_flux(self, dcoll, dd_bdry, u_minus):  # noqa: D102
+    def get_grad_flux(self, dcoll, dd_bdry, kappa_minus, u_minus):  # noqa: D102
         actx = u_minus.array_context
+        kappa_tpair = TracePair(dd_bdry,
+            interior=kappa_minus,
+            exterior=kappa_minus)
         u_tpair = TracePair(dd_bdry,
             interior=u_minus,
             exterior=u_minus)
         normal = actx.thaw(dcoll.normal(dd_bdry))
-        return grad_facial_flux(u_tpair, normal)
+        return grad_facial_flux(kappa_tpair, u_tpair, normal)
 
     def get_diffusion_flux(
-            self, dcoll, dd_bdry, kappa_minus, grad_u_minus):  # noqa: D102
-        actx = grad_u_minus[0].array_context
+            self, dcoll, dd_bdry, kappa_minus, u_minus, grad_u_minus,
+            lengthscales_minus, *, penalty_amount=None):  # noqa: D102
+        actx = u_minus.array_context
         kappa_tpair = TracePair(dd_bdry,
             interior=kappa_minus,
             exterior=kappa_minus)
+        u_tpair = TracePair(dd_bdry,
+            interior=u_minus,
+            exterior=u_minus)
         normal = actx.thaw(dcoll.normal(dd_bdry))
         grad_u_tpair = TracePair(dd_bdry,
             interior=grad_u_minus,
             exterior=(
                 grad_u_minus
                 + 2 * (self.value - np.dot(grad_u_minus, normal)) * normal))
-        return diffusion_facial_flux(kappa_tpair, grad_u_tpair, normal)
+        lengthscales_tpair = TracePair(
+            dd_bdry, interior=lengthscales_minus, exterior=lengthscales_minus)
+        return diffusion_facial_flux(
+            kappa_tpair, u_tpair, grad_u_tpair, lengthscales_tpair, normal,
+            penalty_amount=penalty_amount)
+
+
+class _DiffusionKappa1Tag:
+    pass
+
+
+class _DiffusionKappa2Tag:
+    pass
```

Comment:
I wonder if it would be cleaner to collapse some stuff like this.  The tag classes seem awkward.  What about sthn like:
```suggestion
class _DiffusionKappaCommTag:
     pass
```

Then using `(_DiffusionKappaCommTag, 1)`, `(_DiffusionKappaCommTag, 2)`.   This strategy could help reduce some of the awkward looking creation of many similarly named tags?

Reply:
I ended up removing the duplicated communication, so there's only one of each `DiffusionKappaCommTag`/`DiffusionStateCommTag` now.